numba/misc/numba_sysinfo.py

```diff
@@ -0,0 +1,639 @@
+            cfs_quota = os_spec_info.get(_cfs_quota, '')
+            cfs_period = os_spec_info.get(_cfs_period, '')
+            runtime_amount = cfs_quota / cfs_period
+            os_spec_info[_cfs_restrict] = runtime_amount
+
+    def format_optional():
+        # Memory
+        units = {_mem_total: 1024, _mem_available: 1024}
+        units.update(params.get('units', {}))
+        for k in (_mem_total, _mem_available):
+            digits = ''.join(d for d in os_spec_info.get(k, '') if d.isdigit())
+            os_spec_info[k] = int(digits or 0) * units[k]
+        # Accessible CPUs
+        split = os_spec_info.get(_cpus_allowed, '').split()
+        if split:
+            n = split[-1]
+            n = n.split(',')[-1]
+            os_spec_info[_cpus_allowed] = str(bin(int(n or 0, 16))).count('1')
+        split = os_spec_info.get(_cpus_list, '').split()
+        if split:
+            os_spec_info[_cpus_list] = split[-1]
+
+    try:
+        format()
+        if not _psutil_import:
+            format_optional()
+    except Exception as e:
+        _error_log.append(f'Error (format shell output): {e}')
+
+    # Call OS specific functions
+    os_specific_funcs = {
+        'Linux': {
+            _libc_version: lambda: ' '.join(platform.libc_ver())
+        },
+        'Windows': {
+            _os_spec_version: lambda: ' '.join(
+                s for s in platform.win32_ver()),
+        },
+        'Darwin': {
+            _os_spec_version: lambda: ''.join(
+                i or ' ' for s in tuple(platform.mac_ver()) for i in s),
+        },
+    }
+    key_func = os_specific_funcs.get(os_name, {})
+    os_spec_info.update({k: f() for k, f in key_func.items()})
+    return os_spec_info
+
+
+def get_sysinfo():
+
+    # Gather the information that shouldn't raise exceptions
+    sys_info = {
+        _start: datetime.now(),
+        _machine: platform.machine(),
+        _cpu_name: llvmbind.get_host_cpu_name(),
+        _cpu_count: multiprocessing.cpu_count(),
+        _platform_name: platform.platform(aliased=True),
+        _platform_release: platform.release(),
+        _os_name: platform.system(),
+        _os_version: platform.version(),
+        _python_comp: platform.python_compiler(),
+        _python_impl: platform.python_implementation(),
+        _python_version: platform.python_version(),
+        _numba_env_vars: {k: v for (k, v) in os.environ.items()
+                          if k.startswith('NUMBA_')},
+        _llvm_version: '.'.join(str(i) for i in llvmbind.llvm_version_info),
+        _roc_available: roc.is_available(),
+        _psutil: _psutil_import,
+    }
+
+    # CPU features
+    try:
+        feature_map = llvmbind.get_host_cpu_features()
+    except RuntimeError as e:
+        _error_log.append(f'Error (CPU features): {e}')
+    else:
+        features = sorted([key for key, value in feature_map.items() if value])
+        sys_info[_cpu_features] = textwrap.fill(' '.join(features), 80)
```

Comment:
This one is strangely quite difficult. One my machine, I have 3 lines of features (...is it normal?). Anyway, that wouldn't fit on the same format as the other values. How about indenting? Let me know if you have something specific in mind.

Reply:
I also had a second thought about the start time after changing it from utc to local. I thought it would be better but then I saw that the script is running on the CI. Should we have both local and UTC?

Reply:
RE times, both local and UTC would probably be good.

RE CPU features, don't worry about it if it's a load of hassle!

Reply:
I've reinstated UTC time in 138c60a and fixed the CPU-features display to match the other lines. It looks ok with a longish output, I thought it would be worse.